addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -17,76 +17,166 @@ class StockPickingBatch(models.Model):
         help='Name of the batch transfer')
     user_id = fields.Many2one(
         'res.users', string='Responsible', tracking=True, check_company=True,
+        readonly=True, states={'draft': [('readonly', False)], 'in_progress': [('readonly', False)]},
         help='Person responsible for this batch transfer')
     company_id = fields.Many2one(
         'res.company', string="Company", required=True, readonly=True,
         index=True, default=lambda self: self.env.company)
     picking_ids = fields.One2many(
-        'stock.picking', 'batch_id', string='Transfers',
-        domain="[('company_id', '=', company_id), ('state', 'not in', ('done', 'cancel'))]",
+        'stock.picking', 'batch_id', string='Transfers', readonly=True,
+        domain="[('id', 'in', allowed_picking_ids)]", check_company=True,
+        states={'draft': [('readonly', False)], 'in_progress': [('readonly', False)]},
         help='List of transfers associated to this batch')
+    show_check_availability = fields.Boolean(
+        compute='_compute_move_ids',
+        help='Technical field used to compute whether the check availability button should be shown.')
+    allowed_picking_ids = fields.One2many('stock.picking', compute='_compute_allowed_picking_ids')
+    move_ids = fields.One2many(
+        'stock.move', string="Stock moves", compute='_compute_move_ids')
+    move_line_ids = fields.One2many(
+        'stock.move.line', 'batch_id', string='Stock move lines', readonly=True,
+        states={'draft': [('readonly', False)], 'in_progress': [('readonly', False)]}
+        )
```

Comment:
will this work?
``` python
move_ids = fields.One2many(related=‘picking_ids.move_lines’)
move_line_ids = fields.One2many(related=‘picking_ids.move_lines.move_line_ids’)
```

Reply:
Seems it doesn't work properly...
As `move_ids` will be the move from the first picking in `self.picking_ids`, and as `move_line_ids` will be the move lines from the first move of the first picking.